drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,473 @@
+          measurement on device %#X\n", DEV_ADDR);
+    i2c_acquire(DEV_I2C);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, cmd, 2, 0);
+    i2c_release(DEV_I2C);
+
+    xtimer_remove(continuous_timer);
+
+    return ret;
+}
+
+int8_t SDP3x_soft_reset(const sdp3x_t *dev)
+{
+    int ret = 0;
+
+    DEBUG("[SDP3x] soft_reset: Sending soft reset to all devices\n");
+    i2c_acquire(DEV_I2C);
+    /* General Call Reset */
+    ret = i2c_write_byte(DEV_I2C, 0x00, 0x06, 0);
+    i2c_release(DEV_I2C);
+    /* Wait 20ms for the reset to be processed */
+    xtimer_usleep(20000);
+    DEBUG("[SDP3x] soft_reset: reset done\n");
+    return ret;
+}
+
+int8_t SDP3x_enter_sleep(const sdp3x_t *dev)
+{
+    int ret = 0;
+    uint8_t cmd[2] = { 0x36, 0x77 };
+
+    DEBUG("[SDP3x] enter_sleep: Entering sleep mode on device %#X\n", DEV_ADDR);
+    i2c_acquire(DEV_I2C);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, cmd, 2, 0);
+    i2c_release(DEV_I2C);
+
+    return ret;
+}
+
+int8_t SDP3x_exit_sleep(const sdp3x_t *dev)
+{
+    int ret = 0;
+    uint8_t ptr[1] = { 0 };
+
+    DEBUG("[SDP3x] exit_sleep: Exiting sleep mode on device %#X\n", DEV_ADDR);
+    i2c_acquire(DEV_I2C);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, ptr, 0, 0);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error exiting sleep mode\n");
+        return ret;
+    }
+    xtimer_usleep(2000);
+    ret = i2c_write_bytes(DEV_I2C, DEV_ADDR, ptr, 0, 0);
+    i2c_release(DEV_I2C);
+
+    return ret;
+}
+
+/**
+ *      intern function to read temperature
+ *      @param  dev      device
+ *      @return Temperature value in celsius
+ */
+double _SDP3x_read_temp(const sdp3x_t *dev)
+{
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+        return ret;
+    }
+    DEBUG("[SDP3x] red temperature bare value %i\n", data[1]);
+    return _SDP3x_convert_to_celsius(data[1]);
+}
+
+/**
+ *      intern function to read differential pressure
+ *      @param  dev      device
+ *      @return Pressure value in pascals
+ */
+double _SDP3x_read_pressure(const sdp3x_t *dev)
+{
+    int16_t data[3];
+    uint8_t ret = _SDP3x_read_data(dev, data);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] Error reading date\n");
+        return ret;
+    }
+    return _SDP3x_convert_to_pascal(dev, data[0]);
+}
+
+/** Read measurements
+ *
+ *      Data consists of:
+ *      2 byte Differential Pressure,
+ *      1 byte CRC
+ *      2 byte Temperature,
+ *      1 byte CRC
+ *      2 byte Scale Factor differential pressure
+ *      1 byte CRC
+ *
+ *      @param addr   Address byte 1 = 0x21, 2 = 0x22, 3 = 0x23,
+ *                    else = Last address
+ *      @param data   Data will be stored here
+ *
+ *      @return       0 if data could be read, 1 if CRC-Error
+ */
+static int8_t _SDP3x_read_data(const sdp3x_t *dev, int16_t *data)
+{
+    int ret = 0;
+    uint8_t *readData = calloc(9, sizeof(uint8_t));
```

Comment:
You can just put 9 bytes on the stack
```suggestion
    uint8_t readData[9];
```
the overhead of malloc is much too big for such simple, short lived allocations - and no need to `free()` anything.

Reply:
Resolved in fixup commit 6b544ab8d1ee03eeff471b3d4190e00be1f6e85f